gdtoolkit/parser.py

```diff
@@ -39,8 +13,33 @@ class Indenter(indenter.Indenter):
     CLOSE_PAREN_types = ["RPAR", "RSQB", "RBRACE"]
     INDENT_type = "_INDENT"
     DEDENT_type = "_DEDENT"
-    tab_len = 4  # TODO: guess
-
-
-_parser = None
-_parser_with_metadata_gathering = None
+    # TODO: guess tab length
+    tab_len = 4
+
+
+THIS_DIRECTORY: str = os.path.dirname(__file__)
+# Caching the parser objects for performances
+_PARSER_WITH_METADATA: Tree = Lark.open(
+    os.path.join(THIS_DIRECTORY, "gdscript.lark"),
+    postlex=Indenter(),
+    parser="lalr",
+    start="start",
+    propagate_positions=True,
+)
+_PARSER: Tree = Lark.open(
+    os.path.join(THIS_DIRECTORY, "gdscript.lark"),
+    postlex=Indenter(),
+    parser="lalr",
+    start="start",
+)
```

Comment:
the caching must be done lazily as `Lark.open()` is time-consuming. Having those two loaded upfront will make e.g. `gdlint` loading `_PARSER` which is actually not used by linter. So in other words, the `gdlint` will call `Lark.open()` two times, hence causing the slowdown.

Reply:
Sure. I'll try to refactor it in a class with cached and lazily evaluated properties.